src/e3/anod/qualifiers_manager.py

```diff
@@ -219,6 +229,118 @@ def repr(self, value: str | bool, hash_pool: list[str] | None) -> str:
             return ""
 
 
+class KeySetDeclaration(QualifierDeclaration):
+    # The separator use to distinguish all the element of the list
+    LIST_SEPARATOR = ";"
+
+    def __init__(
+        self,
+        origin: str,
+        name: str,
+        description: str,
+        repr_in_hash: bool = False,
+        repr_name: str | None = None,
+        repr_omit_key: bool = False,
+        default: set[str] | None = None,
+        choices: list[str] | None = None,
+    ) -> None:
+        """Initialize a key-set qualifier declaration.
+
+        :param origin: a string giving the origin of the declaration
+            (used in error messages)
+        :param name: see QualifierDeclaration.__init__
+        :param description: see QualifierDeclaration.__init__
+        :param repr_in_hash: see QualifierDeclaration.__init__
+        :param repr_name: see QualifierDeclaration.__init__
+        :param repr_omit_key: if True discard qualifier name in string representation
+        :param default: default value for the qualifier
+        :param choices: list of valid value for the qualifier
+        """
+        super().__init__(
+            origin=origin,
+            name=name,
+            description=description,
+            repr_in_hash=repr_in_hash,
+            repr_name=repr_name,
+        )
+        self.repr_omit_key = repr_omit_key
+
+        # Check if the default is valid
+        if default is not None and choices is not None:
+            wrong_values = default - set(choices)
+            if wrong_values:
+                choices_str = ", ".join((f"'{choice}'" for choice in choices))
+                wrong_values_str = ", ".join(
+                    (f"'{value}'" for value in sorted(wrong_values))
+                )
+                raise AnodError(
+                    f"{self.origin}: In '{self.name}', default value(s) "
+                    f"({wrong_values_str}) should be in ({choices_str})"
+                )
+
+        self.choices = choices
+        self._default: frozenset[str] | None = (
+            frozenset(default) if default is not None else None
+        )
+
+    @property
+    def default(self) -> str | bool | frozenset[str] | None:
+        """See QualifierDeclaration.default."""
+        return self._default
+
+    def value(self, value: str | bool | frozenset[str]) -> str | bool | frozenset[str]:
```

Comment:
there is no case in which a frozen set is accepted as input

Reply:
I have also remove bool for the same reason. If I missed something, let me know.